crates/order-stream/src/lib.rs

```diff
@@ -415,4 +456,59 @@ mod tests {
 
         assert_eq!(order, db_order.order);
     }
+
+    #[sqlx::test]
+    async fn test_pending_connection_timeout(pool: PgPool) {
+        let anvil = Anvil::new().spawn();
+        let rpc_url = anvil.endpoint_url();
+
+        let ctx =
+            TestCtx::new(&anvil, Digest::from(SET_BUILDER_ID), Digest::from(ASSESSOR_GUEST_ID))
+                .await
+                .unwrap();
+
+        ctx.prover_market
+            .deposit_stake_with_permit(default_allowance(), &ctx.prover_signer)
+            .await
+            .unwrap();
+
+        let config = Config {
+            rpc_url,
+            market_address: *ctx.prover_market.instance().address(),
+            min_balance: parse_ether("2").unwrap(),
+            max_connections: 1,
+            queue_size: 10,
+            domain: "0.0.0.0:8585".parse().unwrap(),
+            bypass_addrs: vec![],
+            ping_time: 20,
+        };
+        let app_state = AppState::new(&config, Some(pool)).await.unwrap();
+        let addr = ctx.prover_signer.address();
+
+        // Test case 1: New connection (vacant entry)
+        let pending_connection = app_state.set_pending_connection(addr).await;
+        assert!(pending_connection, "Should return true for a new connection");
+
+        // Test case 2: Existing connection within timeout (occupied entry, not timed out)
+        let pending_connection = app_state.set_pending_connection(addr).await;
+        assert!(!pending_connection, "Should return false for a connection within timeout");
+
+        // Test case 3: Existing connection that has timed out
+        // Manually set the timestamp to be older than the timeout
+        {
+            let mut pending_connections = app_state.pending_connections.lock().await;
+            let old_time =
+                Instant::now() - (AppState::PENDING_CONNECTION_TIMEOUT + Duration::from_secs(1));
+            pending_connections.insert(addr, old_time);
+        }
+
+        // Now it should allow a new connection since the old one timed out
+        let pending_connection = app_state.set_pending_connection(addr).await;
+        assert!(pending_connection, "Should return true for a timed out connection");
+
```

Comment:
It might be good to have one more case here, just to make sure the above action changed the state in the expected way.
```suggestion

        // Newly set connection should result in pending_connection == false
        let pending_connection = app_state.set_pending_connection(addr).await;
        assert!(!pending_connection, "Should return false for a replaced connection within timeout");

```